python/ray/node.py

```diff
@@ -249,12 +250,25 @@ def _init_temp(self, redis_client):
 
     def get_resource_spec(self):
         """Resolve and return the current resource spec for the node."""
+        resources = {}
+        try:
+            env_string = os.getenv("RAY_RESOURCES")
+            if env_string:
+                resources = json.loads(env_string)
+        except Exception:
+            logger.exception(
+                "Unable to parse the RAY_RESOURCES environment variable using "
+                "json.loads, defaulting to empty dict. Try using a"
+                "format like\n\n"
+                "    RAY_RESOURCES='{\"CustomResource1\": 3, "
+                "\"CustomReseource2\": 2}'")
+
         if not self._resource_spec:
+            resources.update(self._ray_params.resources)
```

Comment:
Are we overwriting if there are same keys from env vars & given custom resources? Why don't we raise errors in that case?

Reply:
Now that we're changing this to non-standard behavior we'll emit a warning.